### DISCUSSION TITLE
How to set two TCP parameters using adb.createSocket in ya-webadb?🥺

### DISCUSSION THREAD

**Discussion by HBLADEH** [yume-chan/ya-webadb#641] [Q&A]
I am using ya-webadb and need some guidance on how to set two TCP parameters using `adb.createSocket("tcp:1234")`.

Specifically, I would like to achieve the same effect as running the command `adb forward tcp:6666 tcp:7777` from the command line.

How should this effect be achieved?🥺🥺

**Comment by HBLADEH** [comment]
I would like to achieve the following effect: using webadb to connect on one computer, then using the `adb forward` command to forward ports, and finally, enabling another computer on the local network to connect using the native` adb connect` command.

**Reply by yume-chan** [reply]
I don't understand what's the relationship between `adb forward` and `adb connect` and how you can use them together.

**Reply by HBLADEH** [reply]
Sorry, I just saw your reply. Previously, I had two computers in a local area network. On PC1, I used ` adb tcpip 7777`, `adb forward tcp:6666 tcp:7777`, and `netsh interface portproxy add v4tov4 listenport=6836 listenaddress=0.0.0.0 connectaddress=127.0.0.1 connectport=6666`. Then, on PC2, I used `adb connect %PC1IP%:6836` to display the phone connected to PC1 on PC2.

Now, I want to change the adb on PC1 to webadb, but I encountered some issues with `adb forward`.😣😣

**Comment by yume-chan** [comment]
`adb forward` is implemented in ADB server. If you are using server-client mode, you might be able to figure out the command string and send it manually using `AdbServerClient#connect`.

Otherwise, you need to re-implement it yourself, by listening at `tcp:6666`, for each incoming socket, call `adb.createSocket("tcp:7777")` and pipe the streams. How to listen on a TCP socket is runtime-dependent.

**Reply by HBLADEH** [reply]
Thank you very much for your reply. I will try this!!!
